### PR TITLE
Link unesco themes to posts

### PR DESCRIPTION
**Pull request by Zama15**
## Add Unesco Feature
unesco_controller.php: Add new unesco controller that inherits from unesco model, added the get_themes method  to load all the themes in the unesco table
db.sql: add more data to insert in the unesco table, now it can be insert a icon with each unesco row
posts/procedure.sql: modfy the save_post procedure to ask for a theme id as argument to make the relation with the unesco table
model/post.php: modify some methods in the post model class

- all: now work with the query constructor from DataBase class instead of the procedure
- find_by_id: same as the all method, now work with the query constructor instead of a procedure
- save: now it is needed a theme id from the unesco table to save the post  

model/unesco.php: added new model, it was created with two initial methods, get_all and get_by_id which work as the name suggest
app.js: add new functios to the app object

- loadUnescoThemes: this make a asyncronys query to load 3 unesco themes initially, have an argument with default value to limit the number of themes it render 
- loadAllUnescoThemes: call the loadUnescoThemes passing as the limit argument with 20, to load all the themes

_side_bar_main.php: remove the static html that use the unesco themes to be load dynamically with an asyncronus method that fetch from the database
post.js: add a function on load to load all the unesco themes as an option in the select input on the new post view
views/posts:

- index: now load in the card the unesco theme that was attached to
- show: same as the index, load the unesco theme that was attached to
- new: created a select input to select the unesco theme as option 

Modify some styles from the cards to make a better load of the posts since the new unesco feature added

## Fixed bugs
index.php: remove the index, this was a previus version now deprecated
posts_controller.php: Add a condition before the render to check if the user is logged in, if it is then render the new method to create a new posts, if not then it redirect to the new method to a new session.
public/index.php: now the navs-main/-side_bar are not generated on the users controller
comment.js: remove from the calling js view the postId and useId variable now being declared here
reaction.js: same as comment.js, removed the variables, so now are declared here
main.js: remove some useless script, since **i don't know what it will be needed in the future i leave it as commented** feel free to eliminated later if this will not be implemented again

## ScreenShots
Screenshot from 2024-05-20 21-34-42
Screenshot from 2024-05-20 21-34-34
Screenshot from 2024-05-20 21-34-27
Screenshot from 2024-05-20 21-34-03

### CODE REVIEW

File: public/resources/js/main.js [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
-        } else {
-            categories.querySelector('.more').textContent = 'View More';
-        }
-        localStorage.setItem('categorie_show', show.classList.contains('show_cat') ? 'show' : 'visible');
-    });
+// document.addEventListener('DOMContentLoaded', function() {
+//     var show = document.getElementById('show_cate');
+//     var categories = document.querySelector('.categories');
+//     categories.addEventListener('click', function() {
+//         show.classList.toggle('show_cat');
```
Comment (MeganPegueros) [inline public/resources/js/main.js:5]: delete lines :)

File: public/resources/js/main.js
Comment (MeganPegueros) [inline public/resources/js/main.js:1]: remove unnecessary lines

**Review by MeganPegueros — APPROVED** [review]

**Review by AngelDiaz591 — APPROVED** [review]

File: public/resources/js/main.js (hunk 1 above)
Comment (Zama15) [reply public/resources/js/main.js:5]: do not worry, that is possible script that could be re-used by another team from the frontend team, so i will leave for now, if i see it is not used, then i will delete it

File: public/resources/js/main.js
Comment (Zama15) [reply public/resources/js/main.js:1]: yep, i make a comment above explain it why i leave

**Pull request merged**